app/apphandlers/embeddedAppHandler.go

```diff
@@ -82,6 +88,7 @@ func (embedded *EmbeddedApp) StartRudderCore(ctx context.Context, options *app.O
 		jobsdb.WithMigrationMode(migrationMode),
 		jobsdb.WithStatusHandler(),
 		jobsdb.WithQueryFilterKeys(jobsdb.QueryFiltersT{}),
+		jobsdb.WithPreBackupHandlers(prebackupHandlers),
```

Comment:
I believe it is a step into the right direction, I would go with it a step further.

Instead of a prebackup handler, I would go with a predelete handler. There you can have multiple implementations:
* Do nothing (no backup at all)
* Backup everything
* Filter backup

It will allow us to test and extend the full backup functionality. 

I understand that this might exceed to scope of this PR, consider the comment non-blocking.

Reply:
I like it, but needs rewriting, thus should treat it as a separate task. I will include one in the backlog